store/src/java/com/zimbra/cs/mailbox/MailboxCacheManager.java

```diff
@@ -0,0 +1,41 @@
+package com.zimbra.cs.mailbox;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+public interface MailboxCacheManager {
+
+    Collection<Integer> getMailboxIds();
+
+    Set<String> getAccountIds();
+
+    Set<Map.Entry<String, Integer>> getMailboxIdsByAccountId();
+
+    int getMailboxCount();
+
+    Integer getMailboxKey(String accountId);
+
+    void removeMailboxId(String accountId);
+
+    Object retrieveFromCache(int mailboxId, boolean trackGC, MailboxManager mailboxManager) throws MailServiceException;
+
+    void cacheMailbox(int mailboxId, MailboxMaintenance maint);
+
+    Object getMailbox(int mailboxId);
+
+    Mailbox cacheMailbox(Mailbox mailbox, MailboxManager mailboxManager);
+
+    void removeMailbox(int mailboxId);
+
+    void cacheAccount(String accountId, int mailboxId);
+
+    int getMailboxCacheSize();
+
+    Collection<Object> getAllLoadedMailboxes();
```

Comment:
Change to `List<Object>`. Why is the generic type `Object` instead of `Mailbox`?

Reply:
changed to `List<Object>`. The reason of returning `Object` instead of `Mailbox`is because the existing `MailboxMap` is returning a Collection of generic objects https://github.com/Zimbra/zm-mailbox/blob/eb42e6de99b3108de2b84f6a4ffc7decdeeb65d7/store/src/java/com/zimbra/cs/mailbox/LocalMailboxCacheManager.java#L279. I though in leaving this as is to not change the behavior of the existing code. Do we need to change it and return `List<Mailbox>` ?